databricks-s3-volume/main.tf

```diff
@@ -2,75 +2,119 @@
 
 // https://docs.databricks.com/administration-guide/multiworkspace/iam-role.html#language-Your%C2%A0VPC,%C2%A0custom
 locals {
-  unity_aws_role_name = "${var.catalog_name}-unity"
+  unity_aws_role_name    = "${var.catalog_name}-unity"
+  iam_role_path          = "/databricks/"
+  databricks_aws_account = "414351767826" # Databricks' own AWS account, not CZI's. See https://docs.databricks.com/en/administration-guide/account-settings-e2/credentials.html#step-1-create-a-cross-account-iam-role
+
+  # catalog and schema use underscores
+  # bucket names use hyphens
 
-  # Create non-hyphenated versions of the catalog and schema names if catalog and/or schema doesnt exist. Else, use the provided names
   catalog_name = var.create_catalog ? replace(var.catalog_name, "-", "_") : var.catalog_name
   schema_name  = var.create_schema ? replace(var.schema_name, "-", "_") : var.schema_name
   volume_name  = replace(var.volume_name, "-", "_")
 
-  path                   = "/databricks/"
-  databricks_aws_account = "414351767826" # Databricks' own AWS account, not CZI's. See https://docs.databricks.com/en/administration-guide/account-settings-e2/credentials.html#step-1-create-a-cross-account-iam-role
-  bucket_name = var.volume_bucket != null ? var.volume_bucket : (
-    var.override_bucket_name != null ? var.override_bucket_name : replace(var.catalog_name, "_", "-") # buckets don't work with underscores
+  volume_bucket_name = var.create_volume_bucket ? replace(var.volume_bucket_name, "_", "-") : var.volume_bucket_name
+  catalog_bucket_name = coalesce(
+    replace(var.catalog_bucket_name, "_", "-"),
+    local.catalog_name
   )
 
-  create_storage_credential = var.create_catalog ? true : (var.create_storage_credential ? true : false)
-
-  # Allow overriding the storage location in case of an existing bucket
-  storage_location = var.override_storage_location != null ? var.override_storage_location : "s3://${local.bucket_name}/${local.schema_name}/${local.volume_name}"
+  create_storage_credentials = var.create_catalog || var.create_storage_credentials
+  volume_storage_location = coalesce(
+    var.volume_storage_location,
+    "s3://${local.volume_bucket_name}/${local.schema_name}/${local.volume_name}"
+  )
 
+  dbx_resource_storage_config = toset([
```

Comment:
Doesn't this mean that we'd always create a catalog bucket and a volume bucket?

Reply:
Ensured that catalog bucket can be created only when specified. Also moved a bunch of iterators to locals for clarity and converted some to maps.